core/src/main/java/org/springframework/security/converter/ObjectConverters.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.converter;
+
+import org.springframework.core.convert.converter.Converter;
+
+import java.net.URI;
+import java.net.URL;
+import java.time.Instant;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * A collection of {@link Converter} implementations
+ * that are capable of converting from {@code Object} to another type.
+ *
+ * @author Joe Grandja
+ * @since 5.2
+ * @see Converter
+ */
+public final class ObjectConverters {
+
+	private ObjectConverters() {
+	}
+
+	/**
+	 * Returns an {@code Object} to {@code String} {@link Converter}.
+	 *
+	 * @return an {@code Object} to {@code String} {@link Converter}
+	 */
+	public static Converter<Object, String> stringConverter() {
```

Comment:
What are your thoughts around

```java
public static final Converter<Object, String> STRING_CONVERTER = ObjectConverters::asString
```

It seems like a constant lookup would perform faster than a method invocation.

Reply:
> It seems like a constant lookup would perform faster than a method invocation

Not sure about this? Do you think the static method approach would be too slow?

Also keep in mind that this follows the same pattern as `RsaKeyConverters.pkcs8()` and `RsaKeyConverters.x509` and being consistent is key.

Reply:
The reason in `RsaKeyConverters` is so that each gets its own instance of `KeyFactory`, but I didn't see any state in the `ObjectConverters`.

Reply:
There is a bit of a tradeoff. Holding static values improves performance a bit as it won't need to be garbage collected. However, it cannot be garbage collected when there continues to be a reference to it, so that means there is additional memory overhead.

Reply:
@rwinch It's not clear to me whether you are requesting changes here? If you are, can you please provide more detail? With the current design, garbage collection won't happen after the class is loaded. For example, using `stringConverter()` as an example, which is `static` and it references the `static` method `asString`. Are you suggesting that `stringConverter()` be changed to non-static, more like a factory method?

Reply:
I didn't really suggest one way or the other, but pointed out the tradeoffs. 

> With the current design, garbage collection won't happen after the class is loaded.

It will happen every time `stringConverter()` is used and the reference is no longer held. For example, here GC could happen as soon as `foo()` completed:

```java
private void foo() {
    Converter<Object, String> toString = stringConverter();
    ...
}
```

If `foo()` is invoked a lot then there is going to be a lot of GC that can happen.

On the flip side `ObjectConverters` creates static members and returns the same instance, there would not be GC happening and multiple invocations of `stringConverter()` would return the same object. This is a plus. However, the static values will never be GC'd which is a tradeoff if we anticipate the values only being used at startup.

Given this is likely to be used outside of startup, I would suggest using the static members as @jzheaux suggested.